File: C++/hsRefCount.cpp
#include "hsRefCount.h"

hsRefCount::hsRefCount() : refCount(1) {  }
hsRefCount::~hsRefCount() { }

int hsRefCount::getRefCount() { return refCount; }

int hsRefCount::UnRef() {
    if (refCount == 1)
        delete this;

    return --refCount;
}

void hsRefCount::Ref() {
    refCount++;
}

File: Java/Plasma/hsRefCount.java
package Plasma;

// Just so everyone knows, this class is completely useless in Java :)

public class hsRefCount {
    protected static int refCount = 1;

    int getRefCount() { return refCount; }
    int removeRef() { return --refCount; }
    void addRef() { refCount++; }
}

